opentelemetry/src/metrics/mod.rs

```diff
@@ -273,22 +273,12 @@ pub trait InstrumentProvider {
     }
 
     /// creates an instrument for recording a distribution of values.
-    fn f64_histogram(
-        &self,
-        _name: Cow<'static, str>,
-        _description: Option<Cow<'static, str>>,
-        _unit: Option<Cow<'static, str>>,
-    ) -> Result<Histogram<f64>> {
+    fn f64_histogram(&self, _builder: HistogramBuilder<'_, f64>) -> Result<Histogram<f64>> {
```

Comment:
this is crucial, and allow the future extensibility we are after.
Agree with this, lets follow up next, by leveraging this to add histogram custom bound support.